src/Common/src/System/Net/Sockets/UnixDomainSocketEndPoint.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Text;
+
+namespace System.Net.Sockets
+{
+    /// <summary>Represents a Unix Domain Socket endpoint as a path.</summary>
+    public sealed class UnixDomainSocketEndPoint : EndPoint
+    {
+        private const int MaxPathLength = 92;   // sockaddr_un.sun_path at http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_un.h.html
+        private const int PathOffset = 2;       // = offsetof(struct sockaddr_un, sun_path). It's the same on Linux and OSX
+        private const int MaxSocketAddressSize = PathOffset + MaxPathLength;
+        private const int MinSocketAddressSize = PathOffset + 2; // +1 for one character and +1 for \0 ending
+        private const AddressFamily EndPointAddressFamily = AddressFamily.Unix;
+
+        private static readonly Encoding s_pathEncoding = Encoding.UTF8;
```

Comment:
Are you sure UTF-8 is appropriate here?  Maybe ASCII?  It's not clear to me from the man page how this should be handled...